ocaml/testsuite/tests/typing-modes/val_modalities.ml

```diff
@@ -549,3 +549,63 @@ module type S' =
 
 (* CR zqian: add tests of recursive modules & include w/ modalties, once
    modules can have modes. *)
+
+module type T = sig @@ portable
+  val foo : 'a -> 'a
+  val bar : 'a -> 'a @@ nonportable
+end
+[%%expect{|
+module type T = sig val foo : 'a -> 'a @@ portable val bar : 'a -> 'a end
+|}]
+
+module type T = sig @@ portable
+  include functor S
+end
+[%%expect{|
+module type T =
+  sig
+    val foo : 'a -> 'a @@ portable
+    val bar : 'a -> 'a @@ portable
+    val baz : 'a -> 'a @@ portable
+  end
+|}]
+
+module type T = sig @@ portable
+  include functor S @@ nonportable
+end
+[%%expect{|
+module type T =
+  sig
+    val foo : 'a -> 'a
+    val bar : 'a -> 'a
+    val baz : 'a -> 'a @@ portable
```

Comment:
I find this quite surprising. This means, I think, that

```
sig
  val f : ... @@ nonportable
end
```

and

```
sig
  val f : ...
end
```

are different -- even though `nonportable` is the default. Is there a design document somewhere that describes this? I want to understand it better. My general philosophy is that having a default meaning when something is omitted (like the portability modality) is great, but that there should always be a way of writing it explicitly if you want. So is there a `val f : ... @@ .` syntax? But how would that know that you want the blank modality for portability specifically?

Reply:
I think I just misunderstood this -- sorry!